authorization/src/main/java/com/terabite/authorization/controller/AuthorizationController.java

```diff
@@ -1,16 +1,28 @@
 package com.terabite.authorization.controller;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.terabite.authorization.Payload;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.terabite.authorization.repository.UserRepository;
+import com.terabite.authorization.service.UserInformation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("/v1/user")
 public class AuthorizationController {
+    @Autowired
+    private UserRepository memberRepository;
+
     @PostMapping("/signup")
-    public Payload userSignupPost() {
-        return new Payload("Reached signup POST");
+    public ResponseEntity<String> userSignupPost(@RequestBody String request) throws JsonProcessingException
+    {
+        ObjectMapper objectMapper = new ObjectMapper();
+        UserInformation userInformation = objectMapper.readValue(request, UserInformation.class);
+        memberRepository.save(userInformation);
+        return ResponseEntity.status(HttpStatus.CREATED).body(String.format("Created user:\n      Name: %s %s\n  Username: %s", userInformation.getFirstName(), userInformation.getLastName(), userInformation.getLogin().getEmail()));
```

Comment:
Strictly speaking, this isn't necessary. I _think_ that spring comes with Jackson, so all you need to do is return an POJO or something, and Jackson will auto serialize it into a json response. It also works in reverse - Jackson will deseralize an request into a POJO. That said, changing it is beyond the scope of this PR. This is just for future reference and work.